sentinelhub/areas.py

```diff
@@ -475,3 +478,171 @@ def _make_split(self):
+
+    :param shape_list: A list of geometrical shapes describing the area of interest
+    :type shape_list: list(shapely.geometry.multipolygon.MultiPolygon or shapely.geometry.polygon.Polygon)
+    :param crs: Coordinate reference system of the shapes in `shape_list`
+    :type crs: CRS
+    :param bbox_size: Physical size in metres of generated bounding boxes. Could be a float or tuple of floats
+    :type bbox_size: float or (float, float)
+    """
+    def __init__(self, shape_list, crs, bbox_size):
+        super(UTMGridSplitter, self).__init__(shape_list, crs)
+
+        self.bbox_size = self._parse_bbox_size(bbox_size)
+
+        self.utm_grid = self._get_overlapping_utm_grid()
+
+        self._make_split()
+
+    @staticmethod
+    def _parse_bbox_size(bbox_size):
+        """ Parses the parameter `bbox_size`. The parameter defines the physical size of BBoxes
+
+        :param bbox_size: The parameter `bbox_size` from class initialization
+        :type bbox_size: int or float or (int, int) or (float, float)
+        :return: A tuple of ints or floats specifying physical size of bounding boxes
+        :rtype: (int, int) or (float, float)
+        :raises: ValueError
+        """
+        if isinstance(bbox_size, (int, float)):
+            return bbox_size, bbox_size
+        if isinstance(bbox_size, (tuple, list)):
+            if len(bbox_size) == 2 and all(isinstance(bbox_dim, (float, int)) for bbox_dim in bbox_size):
+                return bbox_size[0], bbox_size[1]
+            raise ValueError('Content of bbox_size {} must be 2 ints or 2 floats.'.format(bbox_size))
+        raise ValueError('BBox size must be an int or float or a tuple of 2 ints or floats.')
+
+    def _get_overlapping_utm_grid(self):
+        """ Find UTM grid zones overlapping with input area shape
+
+        :return: List of geometries and properties of UTM grid zones overlapping with input area shape
+        """
+        # file downloaded from faculty.baruch.cuny.edu/geoportal/data/esri/world/utmzone.zip
+        utm_grid_filename = os.path.join(os.path.dirname(__file__), '.utmzones.geojson')
+
+        if not os.path.isfile(utm_grid_filename):
+            raise IOError('UTM grid definition file does not exist: %s' % os.path.abspath(utm_grid_filename))
+
+        with open(utm_grid_filename) as utm_grid_file:
+            utm_grid = json.load(utm_grid_file)['features']
+
+        for utm_zone in utm_grid:
+            utm_zone['geometry']['coordinates'] = np.round(np.array(utm_zone['geometry']['coordinates']), 4).tolist()
+
+        utm_geom_list = GeometryCollection([shape(utm_zone['geometry']) for utm_zone in utm_grid])
+        utm_prop_list = [utm_zone['properties'] for utm_zone in utm_grid]
+
+        return [(utm_geom, utm_prop) for utm_geom, utm_prop in zip(utm_geom_list, utm_prop_list)
+                if utm_geom.intersects(self.area_bbox.transform(CRS.WGS84).geometry)]
```

Comment:
I agree to use shape instead of bbox, although should not make a difference in wgs84. Also transforming of area shape will be done in `__init__`

But I would still return only the overlapping utm areas with the shape, as in `make_split` the bboxes are split and checked for interesection. If they are not-prefiltered, the `make_split` function would be applied to each Utm zone or Utm tile, which seems unnecessary

Reply:
changed as suggested. running times are proportional to the complexity of the input shape geometry, looping through zones doesn't spoil performance